baton-maven-plugin/src/main/java/org/technologybrewery/baton/AbstractMigration.java

```diff
@@ -86,6 +112,67 @@ public MigrationSummary execute(FileSet[] fileSets) {
      */
     protected abstract boolean performMigration(File file);
 
+    protected void backupOriginalFile(File originalFile) {
+        if (shouldBackupMigratedOriginalFiles()) {
+            try {
+                String persistentBackupPathBase;
+                if (StringUtils.isNotBlank(backupCustomLocation)) {
+                    persistentBackupPathBase = getBasedir().getAbsolutePath() + "/" + backupCustomLocation;
+                } else {
+                    persistentBackupPathBase = FileUtils.getTempDirectoryPath();
+                }
+
+                String originalPathAsString = Paths.get(originalFile.toURI()).toString();
+                String baseDirectoryFilePath = Paths.get(getBasedir().toURI()).toString();
+                String localPath = originalPathAsString.replace(baseDirectoryFilePath, "");
```

Comment:
A: The Paths API has a `relativize`/`resolve` method that may be more readable.  May also be less readable given we have a mix of `File` and `Path` here.

Reply:
Yeah - my first pass used it, but ended up being less clear and required more processing to remove the "../../.." content once the value returned.